### PR TITLE
Bugfix: endtime in UTC for 3.0.x

### PR DESCRIPTION
**Pull request by AndreasA**
Split of Bugfix from https://github.com/TYPO3-Solr/ext-solr/pull/7 for v3.0.x

### CODE REVIEW

**Comment by AndreasA** [comment]
Did not test the bugfix for v3.0.x

**Comment by irnnr** [comment]
Can you explain what the issue is / what you're trying to solve?

**Comment by AndreasA** [comment]
OK.
Let's say the following the timezone of the server is Europe/Vienna which means currently GMT+2 (with DST).

Now let's say the endtime of a record is set to "2015-07-14 12:01" (server time).
That means I want the record to be removed from solr at "2015-07-14 12:01:00" (GMT+2).
Now, the endtime in solr will be set like this: "2015-07-14T12:01:00Z".

However, as solr actually expects times to be in UTC this is wrong an will result in the record to be removed from solr at "2015-07-14 14:01:00" (server time) instead of at "2015-07-14 12:01:00" (server time).
This is not an issue for "other" time values but for the endtime it is because endtime is set to be used via "DocExpirationUpdateProcessorFactory"

**Comment by irnnr** [comment]
ok yeah, just saw it in #10, thanks!

File: Classes/FieldProcessor/Service.php [hunk 1]
```diff
@@ -66,20 +66,24 @@ public function processDocument(Apache_Solr_Document $document, array $processin
 				}
 
 				switch ($instruction) {
+					case 'timestampToUtcIsoDate':
+						$processor  = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('Tx_Solr_FieldProcessor_TimestampToUtcIsoDate');
+						$fieldValue = $processor->process($fieldValue);
+						break;
 					case 'timestampToIsoDate':
-						$processor  = t3lib_div::makeInstance('Tx_Solr_FieldProcessor_TimestampToIsoDate');
+						$processor  = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('Tx_Solr_FieldProcessor_TimestampToIsoDate');
```
Comment (irnnr) [inline Classes/FieldProcessor/Service.php]: REALLY can't do this for 3.0.1 ;)

Comment (AndreasA) [inline Classes/FieldProcessor/Service.php]: Damn forgot to revert that.
Thanks for noticing :)

**Comment by AndreasA** [comment]
Fix wrong class used for makeInstance

**Comment by irnnr** [comment]
LGTM, thanks a lot!

**Pull request merged**